server/src/internalClusterTest/java/org/elasticsearch/search/retriever/MinimalCompoundRetrieverIT.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.retriever;
+
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryRewriteContext;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.test.AbstractMultiClustersTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.elasticsearch.transport.RemoteClusterAware;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertResponse;
+import static org.hamcrest.Matchers.equalTo;
+
+public class MinimalCompoundRetrieverIT extends AbstractMultiClustersTestCase {
```

Comment:
Tbf this was just a quick write up on how we could test CCS with a compound retriever. If we are to add this, maybe we should enhance it to increase coverage to verify results / what happens if one cluster fails etc. 

Alternatively, as this is outside the context of this PR, we could maybe instead just add a new ctor to `StandardRetrieverBuilder` that would accept a `QueryBuilder` and use it directly in `CCSUsageTelemetryIT` as follows through an anonymous `RetrieverBuilder` class
```
        @Override
        public RetrieverBuilder rewrite(QueryRewriteContext ctx) throws IOException {
            if (ctx.getPointInTimeBuilder() == null) {
                throw new IllegalStateException("PIT is required");
            }
            return new StandardRetrieverBuilder(QueryBuilders.matchAllQuery());
        }
```

Reply:
For this PR I just wanted to fix the code path that is now broken (as it relates to the telemetry) and have the very basic test that ensures it's un-broken. But it's definitely open to followups/improvements. I'm not sure I feel confident enough to mess with `StandardRetrieverBuilder` but if you do the changes there I could definitely update the `CCSUsageTelemetryIT` test.